vermouth/rcsu/contact_map.py

```diff
@@ -12,10 +12,610 @@
+    """
+    take an array with residue level data and repeat the entries over
+    each atom within the residue
+
+    would be nice to do this with list comprehension but I can't work out how
+    to do something like:
+
+    [np.tile(res_dists[i,j],
+             np.where(residues == i)[0].size)
+        for i in range(nresidues)
+        for j in range(nresidues)
+    ]
+
+    to get it correct in the 2 dimensional way we actually need.
+
+    At the moment we only need this function once
+    (to get the residue COGs foreach atom)
+    so it's not too much of a limiting factor, but something to optimise
+    better in future
+
+    """
+    # find out how many residues we have, and how many atoms are in each of them
+    unique_values, counts = np.unique(residues, return_counts=True)
+
+    assert len(unique_values) == nresidues
+
+    out = np.zeros((len(residues), len(residues)))
+    start0 = 0
+    for i, j in zip(unique_values, counts):
+        start1 = 0
+        for k, l in zip(unique_values, counts):
+            target_value = arrin[i, k]
+            out[start0:start0 + j, start1:start1 + l] = target_value.sum()
+            start1 += l
+        start0 += j
+
+    return out
+
+
+def atom2res(arrin, residues, nresidues, norm=False):
+    """
+    take an array with atom level data and sum the entries over within the residue
+    """
+    out = np.array([int(arrin[np.where(residues == i)[0], np.where(residues == j)[0][:, np.newaxis]].sum())
+                    for i in range(nresidues)
+                    for j in range(nresidues)]).reshape((nresidues, nresidues))
+    if norm:
+        out[out > 0] = 1
+
+    return out
+
+
+def bondtype(i, j):
+    maxatomtype = 10
+    assert i >= 1
+    assert i <= maxatomtype
+    assert j >= 1
+    assert j <= maxatomtype
+
+    i -= 1
+    j -= 1
+    # BOND TYPE
+    # Types of contacts:
+    # HB -- 1 -- hydrogen-bond
+    # PH -- 2 -- hydrophobic
+    # AR -- 3 -- aromatic - contacts between aromatic rings
+    # IB -- 4 -- ionic bridge - contacts created by two atoms with different charges
+    # DC -- 5 -- destabilizing contact - contacts which are in general repulsive
+    # OT -- 6 -- denotes negligible other contacts.
+    # 1-HB,2-PH,3-AR,4-IP,5-DC,6-OT
+    btype = np.array([[1, 1, 1, 5, 5, 6, 6, 6, 1, 1],
+                      [1, 5, 1, 5, 5, 6, 6, 6, 1, 5],
+                      [1, 1, 5, 5, 5, 6, 6, 6, 5, 1],
+                      [5, 5, 5, 2, 2, 6, 6, 6, 5, 5],
+                      [5, 5, 5, 2, 3, 6, 6, 6, 5, 5],
+                      [6, 6, 6, 6, 6, 6, 6, 6, 6, 6],
+                      [6, 6, 6, 6, 6, 6, 6, 6, 6, 6],
+                      [6, 6, 6, 6, 6, 6, 6, 6, 6, 6],
+                      [1, 1, 5, 5, 5, 6, 6, 6, 5, 4],
+                      [1, 5, 1, 5, 5, 6, 6, 6, 4, 5]])
+    return btype[i][j]
+
+
+def contact_info(system):
+
+    system = MergeAllMolecules().run_system(system)
+    G = make_residue_graph(system.molecules[0])
```

Comment:
Hmmn. I understand it but I still don't like it. Maybe do the merge in the martinize2 script (or GoPipeLine), and have contact_info run on molecules? I think that'll make life easier later